lib/display/r_raster.c

```diff
@@ -152,7 +155,28 @@ int D_open_driver(void)
 
     init();
 
-    return 0;
+    /* if GRASS_REGION is not defined, a display module can choose to render
+     * for the entire computational region or wait for the monitor to call it
+     * with GRASS_REGION set to the current display extent (what is shown) in
+     * the monitor; the monitor defines GRASS_REGION to the display extent
+     * dynamically, but it's the display module who is responsible to contrain
+     * its rendering within this region to save computational time; if the
+     * display module does not check the return value of D_open_driver(), it
+     * will have to render the same content twice (once by the command line and
+     * once more by the monitor) and it is potentially more intensive
+     * computationally to render for the entire computational region, so it is
+     * recommended to exit from the module if D_open_driver() returns -1; upon
```

Comment:
Rendering according to computation region seems like plain wrong when the expectation is _only_ to pass the command line deeper to the rendering backend. Is it possible that this GRASS_REGION issue is a symptom of a deeper issue in the monitor system?

Reply:
Like I said in https://github.com/OSGeo/grass/issues/3481#issuecomment-1996725006, we used to have an IPC-based (and socket-based) monitor system for UN*X where we didn't have to call display modules **three times**, IIRC, like we do now. Since we moved away from UN*X only systems and to support non-interactive file-based monitors, we don't have a server-client model anymore. Is it a deeper issue in the monitor system? Yes and no... Yes because it's very annoying to deal with this kind of multiple calls to one command and maybe no because it's a necessary evil within the current architecture when you deal with the terminal and GUI using only environment variables.